server/dump.go

```diff
@@ -90,3 +98,52 @@ func (p *Plugin) createDump(ctx context.Context, min, max time.Time, remoteStora
 
 	return zipFileNameRemote, nil
 }
+
+// Metadata is the auxiliary data added to a plugin downloadable.
+// It should contain the required fields and may have it's custom values as well.
+// TODO: move this to mattermost/server/public/pluginapi in the future
+type metadata struct {
+	ServerVersion string         `yaml:"server_version"`
+	ServerID      string         `yaml:"server_id"`
+	LicenseID     string         `yaml:"license_id"`
+	PluginID      string         `yaml:"plugin_id"`
+	Custom        map[string]any `yaml:"custom"`
+}
+
+func (p *Plugin) genarateMetadataForDump(path string, min, max time.Time) (*metadata, error) {
```

Comment:
nit: align naming to the other generateSupportPackage methods in the codebase (e.g. generateSupportPacketYaml) - easier to search for later-on.